File: platforms/documentation/docs/src/snippets/reference/plugin-development/validatePluginsConfig/common/src/main/java/org/example/HelloPlugin.java
package org.example;

import org.gradle.api.Plugin;
import org.gradle.api.Project;

public class HelloPlugin implements Plugin<Project> {

    @Override
    public void apply(Project project) {
        project.getTasks().register("hello", HelloTask.class, task ->
            task.getMessage().convention("Hello from HelloTask")
        );
    }
}

File: platforms/documentation/reference-docs/src/snippets/structuring-builds/composite-builds-basic/kotlin/my-app/my-plugin/src/main/kotlin/com/example/plugin/HelloPlugin.kt
package com.example.plugin

import org.gradle.api.Plugin
import org.gradle.api.Project

class HelloPlugin : Plugin<Project> {
    override fun apply(project: Project) {
        project.tasks.register("helloFromPlugin") {
            group = "example"
            description = "Prints a message from the included plugin build"
            doLast {
                println("Hello from my-plugin! Applied to: ${'$'}{project.path}")
            }
        }
    }
}
